lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -28,11 +28,10 @@ def _enrollments_for_user(self, user):
     def field_dictionary(self, **kwargs):
         """ add course if provided otherwise add courses in which the user is enrolled in """
         field_dictionary = super(LmsSearchFilterGenerator, self).field_dictionary(**kwargs)
-        if not kwargs.get('user'):
-            field_dictionary['course'] = []
-        elif not kwargs.get('course_id'):
-            user_enrollments = self._enrollments_for_user(kwargs['user'])
-            field_dictionary['course'] = [unicode(enrollment.course_id) for enrollment in user_enrollments]
+        if kwargs.get('user'):
```

Comment:
Why these changes?

What are we accomplishing?

Why do we need to modify the default filter if we're implementing our own (that subclasses it)?

Reply:
Returning an empty course list when there's no user interacted with the former [hardcoded search api logic](https://github.com/edx/edx-search/blob/master/search/api.py#L89) such that it ends up just being an empty dictionary, i.e no filter, so it would display all courses. So that was the default behavior. Now that I had to remove the hardcoded `org` filter, if this logic remained, it would then filter on an empty course list and instead return nothing. To maintain the prior default behavior, I updated this logic